File: UI/gymify_mobile/lib/models/payment_intent_start_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment_intent_start_response.g.dart';

@JsonSerializable()
class PaymentIntentStartResponse {
  final String clientSecret;
  final String intentId;
  final int paymentId;
  final double? amount;
  final String? billingPeriod;

  PaymentIntentStartResponse({
    required this.clientSecret,
    required this.intentId,
    required this.paymentId,
    this.amount,
    this.billingPeriod,
  });

  factory PaymentIntentStartResponse.fromJson(Map<String, dynamic> json) =>
      _$PaymentIntentStartResponseFromJson(json);

  Map<String, dynamic> toJson() =>
      _$PaymentIntentStartResponseToJson(this);
}

File: Gymify.Model/ResponseObjects/PaymentIntentStartResponse.cs
namespace Gymify.Model.ResponseObjects
{
    public class PaymentIntentStartResponse
    {
        public string ClientSecret { get; set; } = string.Empty;
        public string IntentId { get; set; } = string.Empty;
        public int PaymentId { get; set; }
        public decimal Amount { get; set; }
        public string BillingPeriod { get; set; } = string.Empty;
    }
}